src/Symfony/Component/DependencyInjection/Tests/ParameterBag/EnvPlaceholderParameterBagTest.php

```diff
@@ -109,4 +109,22 @@ public function testMergeWithDifferentIdentifiersForPlaceholders()
         $this->assertNotEquals($firstPlaceholder, $secondPlaceholder);
         $this->assertCount(2, $merged[$envName]);
     }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage The default value of env parameter "ARRAY" must be string or null, array given.
+     */
+    public function testResolveThrowsOnBadDefaultValue()
+    {
+        $bag = new EnvPlaceholderParameterBag();
+        $bag->get('env(INT)');
+        $bag->set('env(INT)', 2);
+        $bag->resolve();
+        $this->assertSame('2', $bag->all()['env(int)']);
+
+        $bag = new EnvPlaceholderParameterBag();
+        $bag->get('env(ARRAY)');
+        $bag->set('env(ARRAY)', array());
+        $bag->resolve();
```

Comment:
this should be in a separate test. Otherwise, throwing an exception in the first resolve would provide weird failures (saying that the expected message is wrong, while the issue is that an exception was not expected there).

Mixing 2 different tests in the same PHPunit test is a bad idea, as it makes it much harder to make them reliable.